zerver/tests/test_typed_endpoint.py

```diff
@@ -415,6 +426,48 @@ def view3(
         )
         self.assertFalse(result)
 
+        # Not nesting the Annotated type with the ApiParamConfig inside Optional is fine
+        @typed_endpoint
+        def no_nesting(
+            request: HttpRequest,
+            *,
+            bar: Annotated[
+                Optional[str],
+                StringConstraints(strip_whitespace=True, max_length=3),
+                ApiParamConfig("test"),
+            ] = None,
+        ) -> None:
+            raise AssertionError
+
+        with self.assertRaisesMessage(ApiParamValidationError, "test is too long"):
+            call_endpoint(no_nesting, HostRequestMock({"test": "long"}))
```

Comment:
For this kind of test, it might be nice to call it again with a shorter value just to show that it works.